### PR TITLE
New workflow with a wait state and starting a subworkflow

### PR DESCRIPTION
**Pull request by narehayrapetyan**
#### Summary

Adding a new workflow with a choice step to check task status and adding a stepfunction step that starts a workflow

#### Testing

Ran the main workflow and verified that the second workflow ran:

 *  ./gradlew build
 * aws cloudformation package
    --template-file scheduling-manager-workflows/templates/scheduling_manager_workflows.yml
    --output-template-file scheduling-manager-workflows/templates/output.yml
    --s3-bucket us-east-1-blox-scheduler-manager
    aws cloudformation deploy --template-file scheduling-manager-workflows/templates/output.yml --stack-name deploymentworkflows --capabilities CAPABILITY_IAM
*  then ran the workflow from the console with initial input:
    {
    "name": "n",
    "account": "a"
    }

#### Licensing

This contribution is under the terms of the Apache 2.0 License: Yes

### CODE REVIEW

File: scheduling-manager-workflows/src/main/java/com/amazonaws/blox/schedulingmanager/deployment/steps/GetDeploymentData.java [hunk 1]
```diff
@@ -36,7 +36,15 @@ public void handleRequest(InputStream input, OutputStream output, Context contex
       throws IOException {
     log.debug("getDeploymentData lambda");
 
-    final DeploymentInput deploymentInput = encoder.decode(input, DeploymentInput.class);
+    final DeploymentInput deploymentInput;
+    try {
+      deploymentInput = encoder.decode(input, DeploymentInput.class);
+    } catch (final IOException e) {
```
Comment (wjbuys) [inline scheduling-manager-workflows/src/main/java/com/amazonaws/blox/schedulingmanager/deployment/steps/GetDeploymentData.java]: Might make sense to move this repeated stanza into the encoder.

File: scheduling-manager-workflows/src/main/java/com/amazonaws/blox/schedulingmanager/deployment/steps/StartDeployment.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    final String taskWorkflowInputJson = encoder.encode(taskWorkflowInput);
+
+    //TODO: spawn workflows for each task
+    //TODO: don't hardcode the statemachine arn or replace accountid/region
+    final StartExecutionRequest startExecutionRequest =
+        new StartExecutionRequest()
+            .withStateMachineArn(TASK_WF_ARN)
+            .withInput(taskWorkflowInputJson)
+            .withName("StartTaskWorkflow" + UUID.randomUUID().toString());
```
Comment (wjbuys) [inline scheduling-manager-workflows/src/main/java/com/amazonaws/blox/schedulingmanager/deployment/steps/StartDeployment.java]: Don't we have to pass this name on into the task that waits for the execution of this workflow to complete?

**Review by wjbuys — APPROVED** [review]
Looks good, barring the TODOs you've already documented.

File: scheduling-manager-workflows/src/main/java/com/amazonaws/blox/schedulingmanager/deployment/steps/StartDeployment.java (hunk 2 above)
Comment (narehayrapetyan) [reply scheduling-manager-workflows/src/main/java/com/amazonaws/blox/schedulingmanager/deployment/steps/StartDeployment.java]: which task? ECS task?

Comment (wjbuys) [reply scheduling-manager-workflows/src/main/java/com/amazonaws/blox/schedulingmanager/deployment/steps/StartDeployment.java]: Ah, damn, terminology overlap. No, the Stepfunctions Task that waits for the start task workflow(s) to complete. Presumably we want to do something once the task is actually up and running? Or will we just get that notification by looking at the actual state of the cluster?

Comment (narehayrapetyan) [reply scheduling-manager-workflows/src/main/java/com/amazonaws/blox/schedulingmanager/deployment/steps/StartDeployment.java]: No so this workflow just kicks off workflows and completes. The workflows themselves have waits until the task is up. If the workflow fails, the task will be started by the next reconciliation/deployment loop.

File: scheduling-manager-workflows/src/main/java/com/amazonaws/blox/schedulingmanager/deployment/steps/GetDeploymentData.java (hunk 1 above)
Comment (narehayrapetyan) [reply scheduling-manager-workflows/src/main/java/com/amazonaws/blox/schedulingmanager/deployment/steps/GetDeploymentData.java]: good idea

**Pull request merged**